mitodl/mitxonline
# Update how we call the posthog API to include another person identifier to allow for easier targetting

Through the redesign, I've discovered that it's impossible to filter by user in our current setup. This adds user.id to all calls so we can use the django admin user id for a user to target them for feature flags in posthog

# What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/2896

# Description (What does it do?)
Updates both check feature flag functions to accept a unique identifier.
Adds user id to react components/js places where it was missing.
Adds to python calls where it was missing as well.

# How can this be tested?

- Load the app locally.
- Go to your inspect window and remove the ph_ cookie for the app OR use a new browser/private session.
- Navigate around the app as usual. I suggest hitting home and catalog at the very least.
- Go to [Posthog's event viewer](https://app.posthog.com/events) - I have modified this for all users to automatically show person properties of `environment` and `user_id`. If you don't have access OR don't see those columns, I can help.
- Make sure you see events for yourself. `user_id` should match your django user's ID (likely 12, or 3 unless you generated a lot of users in your instance). The should now include `user_id` and `environment` for both python and javascript("web") SDKs.
- Now, go to one of those flags ([mitxonline-new-product-page](https://app.posthog.com/feature_flags/15708) is a good candidate as it is used in a number of places - both js and python as well. Turn the dev flag to 0% of users. This simulates the current state of production.
- Check that you can not see the changes expected (try going to http://mitxonline.odl.local:8013/catalog/ - should get a 404). To see as prod presently is, do these changes on ALL flags.
- Now add a new condition (I already have one in there that I'll leave set to 0%, but this is so you know what I did) - user_id = your_django_user_id - mine is 1, yours might vary, be sure to change this to match yours. environment=dev. Turn this to 100% - This will be how we will turn this on for a specific set of users.
- Check that the state changes and you can see redesign changes under the flag(s) you changed.
